nav2_smac_planner/src/a_star.cpp

```diff
@@ -192,35 +192,81 @@ template<>
 void AStarAlgorithm<Node2D>::setGoal(
   const float & mx,
   const float & my,
-  const unsigned int & dim_3)
+  const unsigned int & dim_3,
+  const GoalHeadingMode & goal_heading_mode)
```

Comment:
```suggestion
  const GoalHeadingMode & /*goal_heading_mode*/)
```

Then you can remove the `void` statement